autogluon/task/text_prediction/modules/basic_prediction.py

```diff
@@ -0,0 +1,427 @@
+            cfg.activation = 'tanh'
+            cfg.normalization = 'layer_norm'
+            cfg.norm_eps = 1e-5
+            cfg.initializer = CfgNode()
+            cfg.initializer.weight = ['xavier', 'uniform', 'avg', 3.0]
+            cfg.initializer.bias = ['zeros']
+        else:
+            raise NotImplementedError
+        return cfg
+
+    def hybrid_forward(self, F, field_proj_features):
+        """
+
+        Parameters
+        ----------
+        field_proj_features
+            List of projection features. All elements must have the same shape.
+
+        Returns
+        -------
+        scores
+            Shape (batch_size,) + out_shape
+        """
+        if len(field_proj_features) == 0:
+            agg_features = field_proj_features[0]
+        else:
+            if self.cfg.agg_type == 'mean':
+                agg_features = F.np.stack(field_proj_features)
+                agg_features = F.np.mean(agg_features, axis=0)
+            elif self.cfg.agg_type == 'concat':
+                agg_features = F.np.concatenate(field_proj_features, axis=-1)
+            else:
+                # TODO(sxjscience) May try to implement more advanced pooling methods for
+                #  multimodal data.
+                raise NotImplementedError
+        scores = self.proj(agg_features)
+        if len(self.out_shape) != 1:
+            scores = F.np.reshape(scores, (-1,) + self.out_shape)
+        return scores
+
+
+@use_np
+class BERTForTabularBasicV1(HybridBlock):
+    """The basic model for tabular classification + regression with
+    BERT (and its variants like ALBERT, MobileBERT, ELECTRA, etc.)
+    as the backbone for handling text data.
+
+    Here, we use the backbone network to extract the contextual embeddings and use
+    another dense layer to map the contextual embeddings to the class scores.
+
+    Input:
+
+    TextField + EntityField --> TextNet -------> TextFeature
+        ...
+    CategoricalField --> CategoricalNet --> CategoricalFeature  ==> AggregateNet --> logits/scores
+        ...
+    NumericalField ----> NumericalNet ----> NumericalFeature
+    """
+    def __init__(self, text_backbone,
+                 feature_field_info,
+                 label_shape=None,
+                 cfg=None,
+                 prefix=None,
+                 params=None):
+        """
+
+        Parameters
+        ----------
+        text_backbone
+            Backbone network for handling the text data
+        feature_field_info
+            The field information of the training data. Each will be a tuple:
+            - (field_type, attributes)
+        label_shape
+            The shape of the label/number of classes. If we need a scalar, it will be an empty tuple "()".
+        cfg
+            The configuration of the network
+        prefix
+        params
+        """
+        super().__init__(prefix=prefix, params=params)
+        if cfg is None:
+            cfg = BERTForTabularBasicV1.get_cfg()
```

Comment:
why?

Reply:
Here, when the arguments is empty, `get_cfg()` will return the default configuration.

Reply:
I am referring to how it is used in the next line (308). It seems like it is redundant to define cfg here:

```
if cfg is None:
    cfg = BERTForTabularBasicV1.get_cfg()
self.cfg = BERTForTabularBasicV1.get_cfg().clone_merge(cfg)
```

Couldn't this just be:

```
self.cfg = BERTForTabularBasicV1.get_cfg()
if cfg is not None:
    self.cfg = BERTForTabularBasicV1.get_cfg().clone_merge(cfg)
```

The clone merge isn't necessary when cfg is None to my understanding.

Reply:
FYI: I updated comment to be cleaner in the ideal code example. Please use this new version instead of original.